server/legalhold/legal_hold.go

```diff
@@ -81,25 +82,50 @@ func (ex *Execution) Execute() (int64, error) {
 // GetChannels populates the list of channels that the Execution needs to cover within the
 // internal state of the Execution struct.
 func (ex *Execution) GetChannels() error {
+	targetUsers, appErr := getUsersForGroups(ex.papi, ex.LegalHold.GroupIDs)
+	if appErr != nil {
+		return appErr
+	}
+
 	for _, userID := range ex.LegalHold.UserIDs {
 		user, appErr := ex.papi.GetUser(userID)
 		if appErr != nil {
 			return appErr
 		}
+		targetUsers = append(targetUsers, user)
+	}
+
+	// keep track of which users have been processed
+	processedUsersList := make(map[string]struct{})
```

Comment:
Gut tells me this is here because `targetUsers` may end up with repeated users coming from different groups or groups + manually specified in the hold?

Reply:
That's exactly right :+1: